### PR TITLE
feat: add lmm to synthetic experiments

### LINKED ISSUE #19: feat: add linear mixed model to abstract
add a synthetic runner that accepts formula and coefficients for a linear mixed model.

### PR DESCRIPTION
**Pull request by younesStrittmatter**
# Description

add linear mixed model synthetic runner

- resolves #19

# Type of change

- **feat**: A new feature

### CODE REVIEW

File: src/autora/experiment_runner/synthetic/abstract/lmm.py [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+            group_var = group_var.strip()
+
+            # Ensure the group_var is in the data
+            if group_var not in experiment_data.columns:
+                raise ValueError(f"Group variable '{group_var}' not found in the data")
+
+            # Process each part of the random effect (intercept and slopes)
+            for part in random_effects_.split('+'):
+                part = part.strip()
+                std_dev = random_effects[group_var].get(part, 0.5)
```
Comment (chadcwilliams) [inline src/autora/experiment_runner/synthetic/abstract/lmm.py:234]: There is a bug here. 

In the formula, you correctly define the intercept as `1`, but then in your examples you define the intercept as `Intercept`.

For example, 
`formula = 'rt ~ 1 + (1|subject) + x1'` 
`random_effects = {'subject': {'Intercept': .1}}`

This line of code is looking for the key `1` in the random_effects dictionary, which does not exist. I am submitting a PR with a quick fix for this.

**Review by chadcwilliams — CHANGES_REQUESTED** [review]
I left a comment about a small bug in this, and have created a PR that fixes the bug.

**Review by hollandjg — APPROVED** [review]
Nice!

**Review by chadcwilliams — APPROVED** [review]
Seems great!

**Pull request merged**